hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -25,30 +301,461 @@ The following table lists the main parameters used throughout the OTP controller
+Otherwise, this signal is tied to all-zeroes.
+
+#### Interfaces to SRAM Scramblers
+
+The interfaces to the FLASH and SRAM scrambling devices follow a req / ack protocol, where the scrambling device first requests a new ephemeral key by asserting the request channel (e.g. `otp_ram_main_key_i`).
+The OTP controller then fetches entropy from CSRNG and derives an ephemeral key using the SRAM_DATA_KEY and the PRESENT scrambling data path as described further below.
+Finally, the OTP controller returns a fresh ephemeral key via the `otp_ram_main_key_o` struct, which completest the req / ack handshake.
+
+Note that this mechanism always works - even if the SRAM_DATA_KEY has not yet been provisioned and locked (via SRAM_KEY_LOCK).
+If SRAM_DATA_KEY has not been locked, an all-zeros value is used to derive the ephemeral key.
 
 
 ## Design Details
 
-# Programmers Guide
+### Block Diagram
+
+**TODO: revamp the diagram below and add more details**
+The following is a high-level block diagram that illustrates everything that has been discussed.
+
+![OTP Controller Block Diagram](otp_controller_block_diagram.svg)
+
+Note the block diagram shows two PRESENT ciphers, although in reality they should be shared between the functions.
+The sharing of PRESENT should be tied into OTP FSM arbitration, whoever has the right to access the OTP IP should also have authority over the cipher block.
+
+### Data Allocation and Packing
+#### Software View
+
+The effective word width of an OTP IP typically depends on a couple of factors, including the redundancy scheme employed.
+For this the design at hand, it is assumed that this native OTP word-width is 16bit.
+For software convenience, however, these details are abstracted and the open-source OTP controller exposes the OTP storage as a linear address space of 32bit words, which is aligned with the machine word size of the Ibex processor.
+Since the OTP IP employs a redundancy mechanism similar to ECC, this implies however that write operations take place at a granularity of 32bit blocks for non-secret and 64bit blocks for secret partitions (due to the scrambling).
+Hence, software is responsible to appropriately pack and program items, since each 32bit location can only be programmed once.
+
+#### Life Cycle View
+
+Since the life cycle partition is the only partition that needs live updates in-field, proper care must be taken to properly encode data in this partition such that incremental updates are possible.
+The life cycle state is hence encoded such that incremental updates to the state are always carried out at the granularity of a 16bit word.
+Further, the life cycle transition counter is encoded such that each stroke consumes a full 16bit word for the same reason.
+
+See [**TODO: add linkto lifecycle spec and state encoding**] for more details on the life cycle encoding.
+
+### Scrambling, Digest and Ephemeral Key Derivation Mechanism
+
+![OTP Digest Mechanism](otp_digest_mechanism.svg)
```

Comment:
image source here for reference (WIP):
https://docs.google.com/drawings/d/17EEsLEzRIrfesEu3bXCWrCBFwk_Lw0Vz4CwOxU4Pjeo